### PR TITLE
[IMP/REF] web: use dialog service to close the dialogs

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/src/core/dialog/dialog_service.js [hunk 1]
```diff
@@ -15,9 +15,15 @@ export const dialogService = {
                 throw new Error(dialogClass.name + " must be a subclass of Dialog");
             }
             const id = ++dialogId;
+            class dialogController extends dialogClass {
```
Comment (aab-odoo) [inline addons/web/static/src/core/dialog/dialog_service.js:18]:
```
class DialogABC extends dialogClass {}
DialogABC.__id = id;
```

Need to find a good name, but not dialogController

File: addons/base_automation/static/tests/base_automation_error_dialog.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ QUnit.module("base_automation", {}, function () {
         const dialogOpened = new Promise((resolve) => {
             env.services.dialog.bus.on("ADD", null, async ({ class: dialogClass, props }) => {
                 assert.equal(props.data.context, errorContext, "Received the correct error context");
-                assert.equal(dialogClass, BaseAutomationErrorDialog, "Should open the correct dialog class");
+                assert.equal(dialogClass.__proto__, BaseAutomationErrorDialog, "Should open the correct dialog class");
```
Comment (aab-odoo) [inline addons/base_automation/static/tests/base_automation_error_dialog.js]: that's a shame. Couldn't we check the content of the dialog instead?

File: addons/web/static/src/core/dialog/dialog.js [hunk 3]
```diff
@@ -37,6 +39,7 @@ export class Dialog extends Component {
         this.size = this.constructor.size;
         this.technical = this.constructor.technical;
         this.title = this.constructor.title;
+        this.__id = null;
```
Comment (aab-odoo) [inline addons/web/static/src/core/dialog/dialog.js:42]: I would have imagined rather something like
```js
Dialog.__id = null; // explain here what the hell this is
```
and 
```js
this.dialogService.close(this.constructor.__id);
```

File: addons/web/static/tests/core/dialog_tests.js [hunk 4]
```diff
@@ -281,30 +293,44 @@ QUnit.module("Components", (hooks) => {
...
-                assert.step(`dialog_id=${id}_closed`);
-                delete this.dialogIds[id];
-            }
         }
         Parent.template = owl.tags.xml`
               <div>
-                <SimpleDialog t-foreach="Object.keys(dialogIds)" t-as="dialogId" t-key="dialogId"
-                  t-on-dialog-closed="_onDialogClosed(dialogId)"
-                  />
+                <MyDialog t-foreach="Object.keys(dialogIds)" t-as="dialogId" t-key="dialogId"/>
```
Comment (aab-odoo) [inline addons/web/static/tests/core/dialog_tests.js:322]: I'm wondering if those tests aren't simply irrelevant now. Maybe we should test the trio dialog/dialogService/dialogContainer together

File: addons/base_automation/static/tests/base_automation_error_dialog.js (hunk 2 above)
Comment (ged-odoo) [reply addons/base_automation/static/tests/base_automation_error_dialog.js]: instanceof

Comment (ged-odoo) [inline addons/base_automation/static/tests/base_automation_error_dialog.js]:
```js
dialogClass.prototype instanceof BaseAutomationErrorDialog.prototype
```

**Comment by ged-odoo** [comment]
robodoo r+

**Pull request closed without merging**